tutorials/scripting/c_sharp/c_sharp_basics.rst

```diff
@@ -388,3 +388,234 @@ The following tools may be used for performance and memory profiling of your man
+    {
+        private int _taskCount = 0;
+        private DateTime start;
+        public override void _Ready()
+        {
+            start = DateTime.Now;
+        }
+    
+        public override void _Process(double delta)
+        {
+
+        }
+
+        // Prints the amount of time since _Ready started, the current thread, and the name of the calling method
+        // It prints this once when DoStuffAsync is first called, then once again after `duration` in seconds
+        private async Task DoStuffAsync(double duration, string methodName)
+        {
+            var taskId = ++_taskCount;
+            PrintCurrentThread($"Task {taskId} started from {methodName}");
+            await Task.Delay(TimeSpan.FromSeconds(duration));
+            PrintCurrentThread($"Task {taskId} completed");
+        }
+    
+        private void PrintCurrentThread(string info)
+        {
+            var timeStamp = DateTime.Now - start;
+            GD.PrintT(timeStamp.ToString(@"mm\:ss\.ff"), $"Thread: {System.Environment.CurrentManagedThreadId}", info);
+        }
+    }
+
+The first option is to start the task through the Task factory.
+
+.. code-block:: csharp
+
+    // This function can be put in a global static class for convenience
+    public static Task StartTaskFromFactory(Func<Task> newTask)
+    {
+        return Task.Factory.StartNew(newTask,
+                                    CancellationToken.None, 
+                                    TaskCreationOptions.None, 
+                                    TaskScheduler.FromCurrentSynchronizationContext());
+    }
+
+    public override void _Ready()
+    {
+        start = DateTime.Now;
+
+        StartTaskFromFactory(async () => await DoStuffAsync(.5, nameof(_Ready)));
+    }
+
+    public override void _Process(double delta)
+    {
+        if (_taskCount < 3)
+            StartTaskFromFactory(async () => await DoStuffAsync(.5, nameof(_Process)));
+    }
+
+The second option is to mark the entry point method as async anyway.
+
+.. code-block:: csharp
+    
+    public override async void _Ready()
+    {
+        start = DateTime.Now;
+        await DoStuffAsync(.5, nameof(_Ready));
+    }
+
+
+    public override async void _Process(double delta)
+    {
+        if (_taskCount < 3)
+            await DoStuffAsync(.5, nameof(_Process));
+    }
+
+Both the manual task starting method and the ``async void`` method 
+behave identically to an equivalent script written in GDScript 
+that uses its version of the ``await`` keyword; 
+the method pauses once it reaches the ``await``-ed method call.
+The game loop will run until the task completes, at which point execution will continue on the main thread.
+
+Let's look at the output from the above code:
+
+.. code-block::
+
+    00:00.00	Thread: 1	Task 1 started from _Ready
+    00:00.02	Thread: 1	Task 2 started from _Process
+    00:00.03	Thread: 1	Task 3 started from _Process
+    00:00.50	Thread: 1	Task 1 completed
```

Comment:
Task 1 should complete before _Process starts.
Not waiting for Task 1 to complete may result in incomlete scene initalization (for example if we were loading something from internet in _Ready function)

Reply:
Fo now solution is to block _Ready thread:
```csharp
public override _Ready() {
  DoStuffAsync().GetAwaiter().GetResult();
  // or
  DoStuffAsync().ConfigureAwait(false).Wait();
}
```

I don't thing it is good to exit _Ready() if the node is not yet ready de-facto. Telling users, who knows programming weakly, to implement `IsNodeInitialized` or `IsInstanceValid` might confuse them and make their games bugged.